app/src/main/java/com/nervousfish/nervousfish/service_locator/NervousFish.java

```diff
@@ -0,0 +1,79 @@
+package com.nervousfish.nervousfish.service_locator;
+
+import android.app.Application;
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.content.ServiceConnection;
+import android.os.IBinder;
+
+import com.nervousfish.nervousfish.modules.pairing.AndroidBluetoothService;
+import com.nervousfish.nervousfish.modules.pairing.PairingWrapper;
+
+/**
+ * The controller of the NervousFish application
+ */
+public final class NervousFish extends Application implements INervousFish {
```

Comment:
If we bind the service to EntryActivity, the service is stopped as soon as we switch to another activity. Of course we can start the activity again by binding it in the new activity, but a restart of the service is often not what we want. The recommended approach (actually, the only approach that's generally used in Android) for making a service run over the lifetime of the application is by letting the only class that exists during the whole lifetime of the application own it: Application